multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/cf/clients/RecentLogsRetriever.java

```diff
@@ -23,13 +24,19 @@ public List<ApplicationLog> getRecentLogsSafely(CloudControllerClient client, St
     }
 
     public List<ApplicationLog> getRecentLogs(CloudControllerClient client, String appName, LogsOffset offset) {
-        List<ApplicationLog> appLogs = client.getRecentLogs(appName);
+        List<ApplicationLog> logsInDescendingOrder = getLogsInDescendingOrder(client, appName, offset);
+        Collections.reverse(logsInDescendingOrder);
+        return logsInDescendingOrder;
+    }
+
+    private List<ApplicationLog> getLogsInDescendingOrder(CloudControllerClient client, String appName, LogsOffset offset) {
+        List<ApplicationLog> appLogsInDescendingOrder = client.getRecentLogs(appName);
         if (offset == null) {
-            return appLogs;
+            return appLogsInDescendingOrder;
         }
-        return appLogs.stream()
-                      .filter(appLog -> isLogNew(appLog, offset))
-                      .collect(Collectors.toList());
+        return appLogsInDescendingOrder.stream()
+                                       .filter(appLog -> isLogNew(appLog, offset))
+                                       .collect(Collectors.toCollection(ArrayList::new));
```

Comment:
Yes, that's the idea of the method. toCollection is used to ensure the creation of a mutable list since the stream API does not require its implementation to return a mutable/immutable list.

Reply:
The reactor-core library, returns ArrayList for sure